packages/contracts-bedrock/contracts/libraries/Encoding.sol

```diff
@@ -130,24 +138,36 @@ library Encoding {
     }
 
     /**
-     * @notice Adds the version to the nonce
+     * @notice Adds a version number into the first two bytes of a message nonce.
+     *
+     * @param _nonce   Message nonce to encode into.
+     * @param _version Version number to encode into the message nonce.
+     *
+     * @return Message nonce with version encoded into the first two bytes.
      */
-    function addVersionToNonce(uint256 _nonce, uint16 _version)
-        internal
-        pure
-        returns (uint256 nonce)
-    {
+    function encodeVersionedNonce(uint240 _nonce, uint16 _version) internal pure returns (uint256) {
+        uint256 nonce;
         assembly {
             nonce := or(shl(240, _version), _nonce)
         }
+        return nonce;
     }
 
     /**
-     * @notice Gets the version out of the nonce
+     * @notice Pulls the version out of a version-encoded nonce.
+     *
+     * @param _nonce Message nonce with version encoded into the first two bytes.
+     *
+     * @return Nonce without encoded version.
+     * @return Version of the message.
      */
-    function getVersionFromNonce(uint256 _nonce) internal pure returns (uint16 version) {
+    function decodeVersionedNonce(uint256 _nonce) internal pure returns (uint240, uint16) {
+        uint240 nonce;
+        uint16 version;
         assembly {
+            nonce := and(_nonce, 0x0000ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff)
```

Comment:
I don't see any test coverage of the nonce value that is returned here?

Reply:
Good point, I can add